build-rust-xcode.sh

```diff
@@ -1,11 +1,22 @@
 #!/bin/bash
 
-if [[ -f "${HOME}/.bash_profile" ]]; then
-    source "${HOME}/.bash_profile"
+# When building from Xcode we want to ensure that `cargo` is in PATH.
+# as a convenience, add the default cargo install location
+export PATH="$PATH:${HOME}/.cargo/bin"
+
+# Users can optionally set cargo path in xi-mac/.env
+if [[ -f "${SRCROOT}/.env" ]]; then
+    source "${SRCROOT}/.env"
+    if ! [ -z "$CARGO_PATH" ]; then
```

Comment:
This one isn't as important and I think this code is fine, but it's usually less error prone to use [[ when possible (since there's already a requirement on bash)